### PR TITLE
[MLIR][TORCH] Add heavydep tests for Bart, BigBird, and GPT2 model

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
This commit adds e2e heavydep tests for Bart, BigBird, and GPT2 model
via AOT.

Signed-Off By: Vivek Khandelwal <vivek@nod-labs.com>

### CODE REVIEW

File: build_tools/torchscript_e2e_heavydep_tests/functorch_inference.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+device = "cpu"
+dtype = torch.float
+
+model = model_type.from_config(config).to(device, dtype=dtype)
+input_ids = torch.randint(0, config.vocab_size, input_size).to(device)
+decoder_ids = torch.randint(0, config.vocab_size, input_size).to(device)
+train_inputs = {"input_ids": input_ids, "labels": decoder_ids}
+
+
+def inference_fn(model, input, labels):
```
Comment (pashu123) [inline build_tools/torchscript_e2e_heavydep_tests/functorch_inference.py:65]: Pass the inference function as `with torch.no grad():` Also remove `.loss.sum().backward()` since we are only intereseted in inference graph.

File: build_tools/torchscript_e2e_heavydep_tests/functorch_inference.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
```
Comment (pashu123) [inline build_tools/torchscript_e2e_heavydep_tests/functorch_inference.py:141]: We can have the same inference function for all these models.

**Review by pashu123 — CHANGES_REQUESTED** [review]

File: build_tools/torchscript_e2e_heavydep_tests/functorch_inference.py (hunk 1 above)
Comment (pashu123) [reply build_tools/torchscript_e2e_heavydep_tests/functorch_inference.py:65]:
```
with torch.no_grad(): 
   return model(**inputs)
```

File: build_tools/torchscript_e2e_heavydep_tests/functorch_inference.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+@register_test_case(module_factory=lambda: ts_module_forw)
+def BartSequenceClassification_basic(module, tu: TestUtils):
+    module.forward(*forw_inputs)
+
+
+# ==============================================================================
+# BigBird Model
+
+config = BigBirdConfig(attention_type="original_full")
+model_type = AutoModelForMaskedLM
```
Comment (silvasean) [inline build_tools/torchscript_e2e_heavydep_tests/functorch_inference.py:92]: Can you make this a function instead of copying the same code every time?

**Review by silvasean — CHANGES_REQUESTED** [review]

**Comment by vivekkhandelwal1** [comment]
I'm closing this since this is now a stale PR.
